pandas/core/indexers.py

```diff
@@ -307,3 +312,62 @@ def check_bool_array_indexer(array: AnyArrayLike, mask: AnyArrayLike) -> np.ndar
     if len(result) != len(array):
         raise IndexError(f"Item wrong length {len(result)} instead of {len(array)}.")
     return result
+
+
+def check_array_indexer(array, indexer) -> np.ndarray:
+    """
+    Check if `indexer` is a valid array indexer for `array`.
+
+    `array` and `indexer` are checked to have the same length, and the
+    dtype is validated. If it is an integer or boolean ExtensionArray, it is
+    checked if there are missing values present, and it is converted to
+    the appropriate numpy array.
+
+    .. versionadded:: 1.0.0
```

Comment:
1.0 or 1.1?

Reply:
1.0 if we're planning to subsume check_bool_array_indexer.

Reply:
> 1.0 if we're planning to subsume check_bool_array_indexer.

Yes, this is replacing `check_bool_array_indexer` which is already in 1.0.0, so we should do the replacement also for 1.0.0